src/controllers/HierarchicalFacetSearchController.ts

```diff
@@ -0,0 +1,38 @@
+import { IDynamicHierarchicalFacet, IDynamicHierarchicalFacetValue } from '../ui/DynamicHierarchicalFacet/IDynamicHierarchicalFacet';
+import { FacetSearchType, IFacetSearchRequest } from '../rest/Facet/FacetSearchRequest';
+import { IFacetSearchResponse } from '../rest/Facet/FacetSearchResponse';
+import { flatten } from 'underscore';
+
+type Path = string[];
+
+export class HierarchicalFacetSearchController {
+  constructor(private facet: IDynamicHierarchicalFacet) {}
+
+  private get ignoredPaths() {
+    return this.flattenPaths(this.facet.values.allFacetValues.map(value => this.getAllPaths(value)));
+  }
+
+  private getAllPaths(value: IDynamicHierarchicalFacetValue): Path[] {
+    return [value.path, ...this.flattenPaths(value.children.map(child => this.getAllPaths(child)))];
+  }
+
+  private flattenPaths(value: Path[][]): Path[] {
+    return flatten(value, true);
+  }
+
+  public search(terms?: string): Promise<IFacetSearchResponse> {
+    const request: IFacetSearchRequest = {
+      field: this.facet.fieldName,
+      type: FacetSearchType.hierarchical,
+      numberOfValues: this.facet.options.numberOfValues,
+      ignorePaths: this.ignoredPaths,
```

Comment:
`ignorePaths` [doesn't work yet](https://coveord.atlassian.net/browse/SEARCHAPI-4666). Adding this line here would remove the selected path and every children that happens to be already displayed under the facet's suggestions.

For instance, in the following picture:
![image](https://user-images.githubusercontent.com/54454747/82817414-f4dbf800-9e6a-11ea-84d9-7d875a2ca576.png)

`ignorePaths` would be:
```ts
[
  ['scala'],
  ['scala', 'syntax'],
  ['scala', 'version']
];
```